scala-thoughts.md

```diff
@@ -0,0 +1,29 @@
+
+###What is/are your favorite language design choice(s) that the designers of Scala made? Why?
+Scala has a lot of flexibility built in to it, which is nice. A lot of methods accept functions which allow you to accomplish a lot of different things with them. This is aided by the fact that Scala's lambda function syntax is compact and allows for quick creation of simple functions.
+
+I also like that Scala's creators didn't just limit it to functional programming (even if it's meant to be used that way). It further increases the flexibility of the language and allows programmers to benefit from being able to use whichever style best solves the task at hand.
+
+Scala's type inference system is nice as well. I didn't have to specify the type of many things when doing this assignment, but Scala inferred value types perfectly. The only exception was when processing a value of type Any with a case statement that specified variable types. If a value has to be of type String to match the case, then Scala should be able to infer that the value is a String, even if the original value was only specified to be of type Any. While the solution doesn't seem to have this problem, it would have been nice if I never had a problem with this in the first place. (I did eventually get things working without having to use casts, but I think Scala still probably has an issue with raising errors about method calls at compile time when you've made sure everything will be fine at run time.)
+
+###What is/are your least favorite language design choice(s)? Why? And why do you think the designers made that / those choice(s)?
+There were a lot of things that confused / irritated me when working with Scala. One thing that was really confusing is that functions can be called without parenthesis. While this is nice, it was initially upsetting because I couldn't call certain functions with parenthesis even if I wanted to.
+
+I also found it confusing that Scala supports mutable variables and data structures if it is supposed to be a functional language. It probably would have been easier to solve some of the problems if I was allowed to use those features, especially since I have a lot of experience in declarative programming.
```

Comment:
I think it's supposed to be a functional language most of the time, except for the cases where you don't want it to be one. It is an interesting contrast with Haskell though. I'm not sure which I prefer.